### ISSUE TITLE
[BUG] Non-allowed modifications and deletions of documents without permission rights

### ISSUE THREAD

**Issue by dominickleppich** [paperless-ngx/paperless-ngx#4073]
### Description

I experienced the following issue in the paperless-ngx Android App, in my opinion the root of the issue lies in paperless-ngx itself or my understanding of the user / access management.

I tried to add screenshots of every relevant step and attached them at the bottom. I don't know how to collapse them to save some space, sorry for this.

I have three users: admin, myself and wife. I have one single group: family. The group has no permissions set on its own. Both myself and wife are added to the group family. I add a new document and assign the owner to admin. 

When I log in to the web interface as myself, the document is not visible, as expected. When I log in to the Android App as myself, the document is also not visible, as expected. Now I add the family group to the documents View > Groups permissions. I did not add the family group to Edit > Groups permissions. I expect, that the new document is now visible for myself. This is the case for both the Web application as well as the Android App. When I navigate to the document in the Web application as myself, I can open the document. The details about the Correspondent, Document type, Storage path and Tags are hidden as 'Private'. I am not able to edit any of those values in the Web application nor can I delete the document, as expected. When I try to bulk edit the document, I also can't change any properties nor delete the document, again as expected. Now I open the Android App as myself and try to navigate to the document. I can see the document and also all its properties (which were hidden as 'Private' in the Web application). I am not able to edit or delete the document in the details screen. The deletion button is grayed out. When I navigate back to the list of documents, press and hold it to select it, I get to the bulk edit screen. In this screen, I am able to modify the properties like Correspondent, Document type, etc. on the document, contrary to my expectation. I have verified the changed storage path in the Web application after reloading the page. Also in the bulk edit mode, I can click the trash icon, confirm the upcoming dialog and permanently delete the document. The webserver log confirms the deletion of the file. As the Web application doesn't allow me these kinds of modifications to a document I should only have reading access to, I suppose this is a bug. I have never entered admin credentials to my Android App. In my understanding, no matter how broken the Android App might be, the webserver should never allow writing or deleting access to this file, no matter from which client the request was made.

Thanks for your great work, I love this project! As this issue is a huge security concern to me, I would like to help solving it as much as I can. If I can add any further information to help track down this issue, I would gladly help :)

![myself-permissions](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/df3554b1-6f9e-489f-bb70-67bf0e55c76c)
![family-permissions](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/1d4343ec-aa37-4b8d-b66e-81d25bbbb353)
![document-family-read-permissions-admin-view](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/1180b9ab-ea98-4d96-ae3e-726341932f05)

![document-details-view-myself](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/445e9879-c11a-4dbb-a1ea-f31eea38131c)

![android-document-details-myself-details-visible](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/742c638c-7013-4182-b56c-434752e1cab9)
![android-document-bulk-edit](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/f660df0e-63a3-4a3b-a93c-b957748f17a0)
![android-document-bulk-edit-change-storage-path](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/f2ce6ace-a9c3-456f-ad0b-9a9d789845eb)
![android-document-bulk-edit-change-storage-path-confirm](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/38baf11d-3ea9-4c14-9e6b-9a1d36c84065)
![android-document-bulk-edit-deletion-confirm](https://github.com/paperless-ngx/paperless-ngx/assets/10048451/f4ba27f9-9597-4d71-b603-b6377b513d5c)

### Steps to reproduce

1. Create a non-admin user and a group the user is member of.
2. Give non-admin user view permissions.
3. Give group no permissions.
4. Add a new document (I tried this by automatically consuming documents from the consume folder or uploading via web application)
5. Assign the document to the owner admin.
6. Give the document view permissions to the created group the non-user is member of.
7. Open the paperless-ngx Android App (From this step downward, the issue should be somehow reproducible by calling the webserver directly I think)
8. Navigate to the document.
9. Press and hold it to open the bulk-edit mode.
10. Change any properties of the document or delete it by clicking on the trash icon.
11. Confirm the change or deletion.

### Webserver logs

```bash
[2023-08-27 13:02:27,302] [DEBUG] [paperless.handlers] Deleted file /usr/src/paperless/media/documents/originals/Invoice/none/2023/2023-08-27_Certificate_Briefmarken.1Stk.17.08.2023_1408_Inbox.pdf.

[2023-08-27 13:02:27,303] [DEBUG] [paperless.handlers] Deleted file /usr/src/paperless/media/documents/archive/Invoice/none/2023/2023-08-27_Certificate_Briefmarken.1Stk.17.08.2023_1408_Inbox.pdf.

[2023-08-27 13:02:27,304] [DEBUG] [paperless.handlers] Deleted file /usr/src/paperless/media/documents/thumbnails/0000524.webp.
```

### Paperless-ngx version

1.17.1

### Host OS

Windows 11 x64 / Android 12

### Installation method

Docker - official image

### Browser

Firefox

### Configuration changes

I configured the docker-compose.env to active polling, set some variables in this context. Nothing I expect to cause this issue however.

**Comment by shamoon** [comment]
#4007

**Comment by dominickleppich** [comment]
Shame on me I haven't realized this was already fixed..

**Comment by shamoon** [comment]
Ha, not at all mate it was just funny timing. Also sorry I actually had a normal human response written that I thought I posted but for some reason the only reply that came through was the one where I linked the PR, ha.

but yes, tldr I agree backend should handle this and we just fixed it like days ago and some of your thread sounds more app-related (which we don’t maintain)

**Comment by dominickleppich** [comment]
Indeed, I guess I have found and set up this project some hours before the updated version. No worries, no bad feelings because of the short message :D

And yes, this issue was only related to the backend. The Android App was the only way I discovered this issue as the web interface didn't show me unallowed modifications as far as I tried it. But thanks for solving my issue 3 days before encountering it, never got a faster support!
